File: day2/1.cpp
#include <fstream>
#include <vector>
#include <string>
#include <ranges>

int main() {
    std::string line; std::ifstream in("data.txt");

    std::getline(in, line);

    auto nums = line
    | std::views::split(',')
    | std::views::transform([](auto&& subrange) { return std::stoi(std::string(subrange.begin(), subrange.end())); })
    | std::ranges::to<std::vector<int>>();

    for (int ip{}; ip < nums.size(); ip += 4) {
        int& instr = nums[ip];
        
        if (instr == 99) break;

        int& lhs_loc = nums[ip + 1];
        int& rhs_loc = nums[ip + 2];
        int& res_loc = nums[ip + 3]; 
        
        if (instr == 1) nums[res_loc] = nums[lhs_loc] + nums[rhs_loc];
        else if (instr == 2) nums[res_loc] = nums[lhs_loc] * nums[rhs_loc];
    }

    printf("%d", nums.front());
}

File: day2/1.py
with open("data.txt", 'r') as file:
    data = list(map(int, file.read().split(',')))

for i in range(0, len(data), 4):
    if data[i] == 99: break
    elif data[i] == 1: data[data[i + 3]] = data[data[i+1]] + data[data[i + 2]]
    elif data[i] == 2: data[data[i + 3]] = data[data[i+1]] * data[data[i + 2]]
    else: continue

print(data[0])